conans/client/command.py

```diff
@@ -821,31 +822,53 @@ def search(self, *args):
         parser.add_argument('--table', action=OnceArgument,
                             help="Outputs html file with a table of binaries. Only valid for a "
                             "reference search")
+        parser.add_argument("-j", "--json", default=None, action=OnceArgument,
+                            help='json file path where the search information will be written to')
         args = parser.parse_args(*args)
 
         try:
             reference = ConanFileReference.loads(args.pattern_or_reference)
             if "*" in reference:
-                # Fixes a version with only a wilcard (valid reference) but not real reference
+                # Fixes a version with only a wildcard (valid reference) but not real reference
                 # e.j: conan search lib/*@lasote/stable
                 reference = None
         except (TypeError, ConanException):
             reference = None
 
-        if reference:
-            ret = self._conan.search_packages(reference, query=args.query, remote=args.remote,
-                                              outdated=args.outdated)
-            ordered_packages, reference, recipe_hash, packages_query = ret
-            self._outputer.print_search_packages(ordered_packages, reference, recipe_hash,
-                                                 packages_query, args.table)
-        else:
-            if args.table:
-                raise ConanException("'--table' argument can only be used with a reference")
+        info = {"error": False, "results": []}
+        cwd = os.getcwd()
 
-            refs = self._conan.search_recipes(args.pattern_or_reference, remote=args.remote,
-                                              case_sensitive=args.case_sensitive)
-            self._check_query_parameter_and_get_reference(args.pattern_or_reference, args.query)
-            self._outputer.print_search_references(refs, args.pattern_or_reference, args.raw)
+        try:
+            if reference:
+                info = self._conan.search_packages(reference, query=args.query, remote=args.remote,
+                                                   outdated=args.outdated)
+                # search is done for one reference and one remote
+                self._outputer.print_search_packages(info["results"], reference, args.query,
+                                                     args.table)
+            else:
+                if args.table:
```

Comment:
Maybe the --table and --json args are totally incompatible, shouldn't get a json captured exception but a command exception